neuro_san/internals/graph/persistence/registry_manifest_restorer.py

```diff
@@ -129,11 +128,36 @@ def restore(self, file_reference: str = None):
                 file_of_class = FileOfClass(manifest_file)
                 manifest_dir: str = file_of_class.get_basis()
                 registry_restorer = AgentToolRegistryRestorer(manifest_dir)
-                tool_registry: AgentToolRegistry = registry_restorer.restore(file_reference=use_key)
+                try:
+                    tool_registry: AgentToolRegistry = registry_restorer.restore(file_reference=use_key)
+                except FileNotFoundError as exc:
+                    self.logger.error("Failed to restore registry item %s - %s", use_key, str(exc))
+                    tool_registry = None
                 if tool_registry is not None:
                     tool_name: str = Path(use_key).stem
                     tool_registries[tool_name] = tool_registry
                 else:
                     self.logger.error("manifest registry %s not found in %s", use_key, manifest_file)
 
         return tool_registries
+
+    # pylint: disable=too-many-locals
+    def restore(self, file_reference: str = None) -> Dict[str, AgentToolRegistry]:
+        """
+        :param file_reference: The file reference to use when restoring.
+                Default is None, implying the file reference is up to the
+                implementation.
+        :return: a built map of agents tools registries
+        """
+        if file_reference is not None:
+            return self.restore_from_files([file_reference])
+
```

Comment:
Added missing implementation for a case when we restore from directly specified file path.